src/ipfs-server.ts

```diff
@@ -0,0 +1,75 @@
+import IPFS from 'ipfs'
+import HttpApi from 'ipfs-http-server'
+import HttpGateway from 'ipfs-http-gateway'
+
+import dagJose from 'dag-jose'
+// @ts-ignore
+import multiformats from 'multiformats/basics'
+// @ts-ignore
+import legacy from 'multiformats/legacy'
+
+import { createRepo } from 'datastore-s3'
+import HealthcheckServer from "./healthcheck-server"
+
+const IPFS_PATH = process.env.IPFS_PATH || 'ipfs'
+const IPFS_S3_REPO_ENABLED = 'IPFS_S3_REPO_ENABLED' in process.env? process.env.IPFS_S3_REPO_ENABLED : false
+
+const { AWS_BUCKET_NAME } = process.env
+const { AWS_ACCESS_KEY_ID } = process.env
+const { AWS_SECRET_ACCESS_KEY } = process.env
+
+const IPFS_SWARM_TCP_PORT = process.env.IPFS_SWARM_TCP_PORT || 4011
+const IPFS_SWARM_WS_PORT = process.env.IPFS_SWARM_WS_PORT || 4012
+
+const IPFS_API_PORT = process.env.IPFS_API_PORT || 5011
+const IPFS_GATEWAY_PORT = process.env.IPFS_GATEWAY_PORT || 9011
+
+export default class IPFSServer {
+
+    /**
+     * Start js-ipfs instance with dag-jose enabled
+     */
+    static async start(): Promise<void> {
+        const repo = IPFS_S3_REPO_ENABLED ? createRepo({
+            path: IPFS_PATH,
+        }, {
+            bucket: AWS_BUCKET_NAME,
+            accessKeyId: AWS_ACCESS_KEY_ID,
+            secretAccessKey: AWS_SECRET_ACCESS_KEY,
+        }) : null
+
+        // setup dag-jose codec
+        multiformats.multicodec.add(dagJose)
+        const format = legacy(multiformats, dagJose.name)
+
+        const ipfs = await IPFS.create({
+            repo,
+            ipld: { formats: [format] },
+            libp2p: {
+                config: {
+                    dht: {
+                        enabled: true,
+                        clientMode: true,
```

Comment:
I don't think we want client mode right? Iirc it outsources dht requests to other nodes?

Reply:
yep, I think you're right 👍